chips/sam4l/src/adc.rs

```diff
@@ -408,12 +447,19 @@ impl hil::adc::Adc for Adc {
             self.active.set(true);
             self.continuous.set(true);
 
+            let trgsel;
+            if self.cpu_clock.get() {
+                trgsel = 1; // internal timer trigger
```

Comment:
Why not just always use the internal timer? Per the manual (page 1002, section 38.6.14) if the itimer is set to 0, a trigger will occur on each GCLK cycle, which would hopefully allow you to collapse your logic into a single case based on `adc_clk_freq`.